jupyterhub/metrics.py

```diff
@@ -100,11 +100,29 @@ def from_status(cls, status):
             return cls.running
         return cls.stopped
 
+for s in ServerPollStatus:
+    SERVER_POLL_DURATION_SECONDS.labels(status=s)
+
+
+
+SERVER_STOP_DURATION_SECONDS = Histogram(
+    'server_stop_seconds',
+    'time taken for server stopping operation',
+    ['status'],
+)
+
+class ServerStopStatus(Enum):
+    """
+    Possible values for 'status' label of SERVER_STOP_DURATION_SECONDS
+    """
+    stopped = 'stopped'
+    pending = 'pending'
```

Comment:
The only status fields for stop, I think, are 'success' and 'failure'. The failure status should be set in the event of a failure.